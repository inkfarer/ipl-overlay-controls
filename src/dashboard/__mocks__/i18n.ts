import i18next from 'i18next';
import { config } from '@vue/test-utils';

jest.mock('i18next-vue', () => ({
    __esModule: true,
    useTranslation: () => ({ t: i18next.t })
}));

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
    config.global.mocks = {
        $t: i18next.t
    };
});
