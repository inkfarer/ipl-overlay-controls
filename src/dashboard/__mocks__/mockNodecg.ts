import Mock = jest.Mock;

export type ReplicantChangeHandler = (newValue?: unknown, oldValue?: unknown) => void;
export type MockReplicant = { value: unknown };
type OnFunction = (event: string, handler: ReplicantChangeHandler) => void;

export class MockNodecg {
    replicants: {[key: string]: MockReplicant};
    listeners: {[key: string]: ReplicantChangeHandler};
    sendMessage: Mock

    constructor() {
        this.replicants = {};
        this.listeners = {};
        this.sendMessage = jest.fn();
    }

    init(): void {
        const self = this;
        window.nodecg = {
            // @ts-ignore: Just a mock
            Replicant(name: string) {
                const replicantValue: { value: unknown, on: OnFunction } = {
                    value: undefined,
                    on: (event: string, handler: ReplicantChangeHandler) => {
                        self.listeners[name] = handler;
                    }
                };
                self.replicants[name] = replicantValue;
                return replicantValue;
            },
            sendMessage: self.sendMessage,
            // @ts-ignore: Just a mock
            log: {
                info: jest.fn()
            }
        };

        // @ts-ignore: It's a mock.
        window.NodeCG = {
            waitForReplicants: jest.fn().mockResolvedValue(true)
        };
    }
}
