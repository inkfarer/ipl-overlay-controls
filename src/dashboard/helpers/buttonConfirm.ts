/*
buttonConfirm

After importing this script to a dashboard panel, add the attribute 'data-requires-confirmation' to any button.
The button will now ask for confirmation when it is first clicked.
When it is clicked a second time, a 'confirm' event is sent by it, which can be listened for by scripts.

Example:

HTML:
    <button id="dangerous-action-button" data-requires-confirmation>DO A THING</button>
JS:
    document.getElementById('dangerous-action-button').addEventListener('confirm', () => {
        // do something
    });
*/

window.addEventListener('DOMContentLoaded', () => {
    function addListener(btn: HTMLButtonElement) {
        if (btn.dataset.hasConfirmListener !== 'true') {
            btn.addEventListener('click', e => {
                const target = e.target as HTMLButtonElement;

                function resetButton() {
                    target.dataset.isClicked = 'false';
                    target.innerText = target.dataset.originalText;
                    target.style.backgroundColor = '';
                }

                if (target.dataset.isClicked === 'true') {
                    target.dispatchEvent(new Event('confirm'));
                    resetButton();
                } else {
                    target.dataset.isClicked = 'true';
                    target.dataset.originalText = target.innerText;
                    target.style.backgroundColor = 'var(--red)';
                    target.innerText = 'are you sure?';

                    setTimeout(() => {
                        resetButton();
                    }, 5000);
                }
            });

            btn.dataset.hasConfirmListener = 'true';
        }
    }

    const buttons = document.querySelectorAll('button[data-requires-confirmation]');
    buttons.forEach(addListener);

    const observer = new MutationObserver(mutations => {
        mutations.forEach(mutation => {
            mutation.addedNodes.forEach(node => {
                if (node.nodeType === Node.ELEMENT_NODE
                    && node.nodeName === 'BUTTON'
                    && (node as HTMLElement).dataset.requiresConfirmation) {

                    const btn = node as HTMLButtonElement;
                    addListener(btn);
                }
            });
        });
    });

    observer.observe(document.body, {
        subtree: true,
        childList: true
    });
});
