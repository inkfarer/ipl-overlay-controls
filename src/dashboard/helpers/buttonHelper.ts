import { colors } from '../styles/colors';

export function setToggleButtonDisabled(
    toggleTrue: HTMLButtonElement,
    toggleFalse: HTMLButtonElement,
    state: boolean
): void {
    toggleTrue.disabled = state;
    toggleFalse.disabled = !state;
}

export function addChangeReminder(
    elements: HTMLElement[] | NodeListOf<HTMLElement>,
    updateButton: HTMLButtonElement
): void {
    elements.forEach(element => {
        if (!element.tagName) return;

        let event;
        if (element.tagName.toLowerCase() === 'input') event = 'input';
        else if (element.tagName.toLowerCase() === 'select') event = 'change';
        else return;

        element.addEventListener(event, e => {
            const target = e.target as HTMLElement;
            target.dataset.edited = 'true';
            updateButton.style.backgroundColor = colors.red;
        });
    });

    updateButton.addEventListener('click', () => {
        elements.forEach(elem => {
            elem.dataset.edited = 'false';
        });
        updateButton.style.backgroundColor = colors.blue;
    });
}
