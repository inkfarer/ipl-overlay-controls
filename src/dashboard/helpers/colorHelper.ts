export function getContrastingTextColor(backgroundColor: string, lightColor = 'white', darkColor = '#333'): string {
    const { r, g, b } = hexToRbg(backgroundColor);
    const yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000;
    return (yiq >= 128) ? darkColor : lightColor;
}

export function hexToRbg(color: string): { r: number, g: number, b: number } {
    color = color.replace('#', '');
    return {
        r: parseInt(color.substr(0, 2), 16),
        g: parseInt(color.substr(2, 2), 16),
        b: parseInt(color.substr(4, 2), 16)
    };
}
