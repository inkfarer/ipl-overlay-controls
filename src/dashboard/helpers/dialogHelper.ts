import { NodecgDialog } from '../types/dialog';

export function closeDialog(dialogName: string): void {
    (nodecg.getDialog(dialogName) as NodecgDialog).close();
}
