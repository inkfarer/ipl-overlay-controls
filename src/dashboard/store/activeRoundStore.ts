import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
