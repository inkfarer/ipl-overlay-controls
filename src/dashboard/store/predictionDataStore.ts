import { NodeCGBrowser } from 'nodecg/browser';
import { Prediction, PredictionStore } from 'schemas';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { defineStore } from 'pinia';

const predictionStore = nodecg.Replicant<PredictionStore>('predictionStore');

export const predictionReps = [ predictionStore ];

export interface PredictionDataStore {
    predictionStore: PredictionStore;
}

export const usePredictionDataStore = defineStore('prediction', {
    state: () => ({
        predictionStore: null,
    } as PredictionDataStore),
    actions: {
        async lockPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.LOCKED,
                [ PredictionStatus.ACTIVE ]);
        },
        async cancelPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.CANCELED,
                [ PredictionStatus.ACTIVE, PredictionStatus.LOCKED ]);
        },
        async resolvePrediction({ winningOutcomeIndex }: { winningOutcomeIndex: number }) {
            if (winningOutcomeIndex > 1 || winningOutcomeIndex < 0) {
                throw new Error(`Cannot resolve prediction with outcome index ${winningOutcomeIndex}.`);
            } else if (!this.predictionStore.currentPrediction?.id) {
                throw new Error('No prediction to resolve.');
            } else if (this.predictionStore.currentPrediction.status !== PredictionStatus.LOCKED ) {
                throw new Error('Can only resolve a locked prediction.');
            }

            return nodecg.sendMessage('patchPrediction', {
                id: this.predictionStore.currentPrediction.id,
                status: PredictionStatus.RESOLVED,
                winning_outcome_id: this.predictionStore.currentPrediction.outcomes[winningOutcomeIndex].id
            });
        },
        async createPrediction(
            { title, teamAName, teamBName, duration }:
                { title: string, teamAName: string, teamBName: string, duration: number }
        ) {
            if (
                [PredictionStatus.ACTIVE, PredictionStatus.LOCKED]
                    .includes(this.predictionStore.currentPrediction?.status as PredictionStatus)
            ) {
                throw new Error('An unresolved prediction already exists.');
            }

            return nodecg.sendMessage('postPrediction', {
                title,
                outcomes: [
                    { title: teamAName },
                    { title: teamBName }
                ],
                prediction_window: duration
            });
        },
        async reconnect() {
            return nodecg.sendMessage('reconnectToRadiaSocket');
        }
    }
});

async function patchPrediction(
    currentPrediction: Prediction,
    status: PredictionStatus,
    allowedForStatuses: PredictionStatus[]
): Promise<void> {
    if (!currentPrediction || !allowedForStatuses.includes(currentPrediction?.status as PredictionStatus)) {
        throw new Error(`Cannot set prediction status at this time. Status must match one of [${allowedForStatuses.join(', ')}]`);
    }

    return nodecg.sendMessage('patchPrediction', {
        id: currentPrediction.id,
        status: status,
    });
}
