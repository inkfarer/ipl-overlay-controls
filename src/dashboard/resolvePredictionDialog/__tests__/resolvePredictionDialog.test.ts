import ResolvePredictionDialog from '../resolvePredictionDialog.vue';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { config, flushPromises, mount } from '@vue/test-utils';
import { useActiveRoundStore } from '../../store/activeRoundStore';
import { GameWinner } from 'types/enums/gameWinner';
import { mockDialog, mockGetDialog } from '../../__mocks__/mockNodecg';
import { closeDialog } from '../../helpers/dialogHelper';
import { PlayType } from 'types/enums/playType';
import { createTestingPinia, TestingPinia } from '@pinia/testing';
import { usePredictionDataStore } from '../../store/predictionDataStore';
import { IplButton, IplDialogTitle } from '@iplsplatoon/vue-components';

jest.mock('../../helpers/dialogHelper');

describe('ResolvePredictionDialog', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        FontAwesomeIcon: true,
        IplDialogTitle: true,
        IplErrorDisplay: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();

        useActiveRoundStore().$state = {
            activeRound: {
                teamA: {
                    score: 0,
                    id: '123123',
                    name: 'First Team',
                    showLogo: true,
                    players: null,
                    color: null
                },
                teamB: {
                    score: 2,
                    id: '345345',
                    name: 'Second Team',
                    showLogo: false,
                    players: null,
                    color: null
                },
                activeColor: {
                    categoryName: 'Ranked Modes',
                    index: 0,
                    title: 'coolest color',
                    isCustom: false,
                    clrNeutral: '#222'
                },
                match: {
                    id: '01010',
                    name: 'Rad Match',
                    isCompleted: false,
                    type: PlayType.PLAY_ALL
                },
                games: [
                    {
                        winner: GameWinner.BRAVO,
                        stage: 'Blackbelly Skatepark',
                        mode: 'Rainmaker',
                        color: {
                            index: 2,
                            title: 'Cool Color',
                            clrA: '#123123',
                            clrB: '#345345',
                            clrNeutral: '#00FF00',
                            categoryName: 'Cool Colors',
                            isCustom: false,
                            colorsSwapped: false
                        }
                    },
                    {
                        winner: GameWinner.BRAVO,
                        stage: 'MakoMart',
                        mode: 'Tower Control',
                        color: {
                            index: 0,
                            title: 'Cool Color',
                            clrA: '#837693',
                            clrB: '#206739',
                            clrNeutral: '#00AA00',
                            categoryName: 'Custom Color',
                            isCustom: true,
                            colorsSwapped: true
                        }
                    },
                    {
                        winner: GameWinner.NO_WINNER,
                        stage: 'Camp Triggerfish',
                        mode: 'Splat Zones'
                    },
                ],
            },
            swapColorsInternally: false
        };

        usePredictionDataStore().$state = {
            predictionStore: {
                status: {
                    socketOpen: true,
                    predictionsEnabled: true
                },
                currentPrediction: {
                    id: 'prediction123',
                    broadcasterId: 'ipl',
                    broadcasterName: 'IPL',
                    broadcasterLogin: 'eye pee el',
                    title: 'Who will win?',
                    outcomes: [
                        {
                            id: 'outcome-1',
                            title: 'First Team',
                            users: 5,
                            pointsUsed: 10000,
                            topPredictors: [],
                            color: 'BLUE'
                        },
                        {
                            id: 'outcome-2',
                            title: 'Second Team',
                            users: 1,
                            pointsUsed: 1,
                            topPredictors: [],
                            color: 'PINK'
                        }
                    ],
                    duration: 60,
                    status: PredictionStatus.ACTIVE,
                    creationTime: '2020',
                }
            }
        };
    });

    it('matches snapshot with missing prediction data', () => {
        const predictionDataStore = usePredictionDataStore();
        predictionDataStore.predictionStore.currentPrediction = undefined;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the prediction is not locked', () => {
        const predictionDataStore = usePredictionDataStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.RESOLVED;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the current round is not completed', () => {
        const predictionDataStore = usePredictionDataStore();
        const activeRoundStore = useActiveRoundStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        activeRoundStore.activeRound.match.isCompleted = false;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the winner name cannot be determined', () => {
        const predictionDataStore = usePredictionDataStore();
        const activeRoundStore = useActiveRoundStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        activeRoundStore.activeRound.teamA.score = 1;
        activeRoundStore.activeRound.teamB.score = 1;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the winner cannot be determined automatically', () => {
        const predictionDataStore = usePredictionDataStore();
        const activeRoundStore = useActiveRoundStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        activeRoundStore.activeRound.teamA.score = 1;
        activeRoundStore.activeRound.teamA.name = 'unknown team a';
        activeRoundStore.activeRound.teamB.score = 2;
        activeRoundStore.activeRound.teamB.name = 'unknown team b';
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the winner is determined as team A', () => {
        const predictionDataStore = usePredictionDataStore();
        const activeRoundStore = useActiveRoundStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        activeRoundStore.activeRound.teamA.score = 2;
        activeRoundStore.activeRound.teamB.score = 1;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when the winner is determined as team B', () => {
        const predictionDataStore = usePredictionDataStore();
        const activeRoundStore = useActiveRoundStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        activeRoundStore.activeRound.teamA.score = 1;
        activeRoundStore.activeRound.teamB.score = 2;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('handles first outcome being resolved and closes dialog when it completes', async () => {
        const predictionDataStore = usePredictionDataStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });
        predictionDataStore.resolvePrediction = jest.fn().mockResolvedValue({});

        wrapper.getComponent<typeof IplButton>('[data-test="resolve-outcome-1-button"]').vm.$emit('click');
        await flushPromises();

        expect(predictionDataStore.resolvePrediction).toHaveBeenCalledWith({ winningOutcomeIndex: 0 });
        expect(mockGetDialog).toHaveBeenCalledWith('resolvePredictionDialog');
        expect(mockDialog.close).toHaveBeenCalledTimes(1);
    });

    it('handles second outcome being resolved and closes dialog when it completes', async () => {
        const predictionDataStore = usePredictionDataStore();
        predictionDataStore.predictionStore.currentPrediction.status = PredictionStatus.LOCKED;
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });
        predictionDataStore.resolvePrediction = jest.fn().mockResolvedValue({});

        wrapper.getComponent<typeof IplButton>('[data-test="resolve-outcome-2-button"]').vm.$emit('click');
        await flushPromises();

        expect(predictionDataStore.resolvePrediction).toHaveBeenCalledWith({ winningOutcomeIndex: 1 });
        expect(mockGetDialog).toHaveBeenCalledWith('resolvePredictionDialog');
        expect(mockDialog.close).toHaveBeenCalledTimes(1);
    });

    it('closes dialog on dialog title close event', () => {
        const wrapper = mount(ResolvePredictionDialog, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent<typeof IplDialogTitle>('ipl-dialog-title-stub').vm.$emit('close');

        expect(closeDialog).toHaveBeenCalledWith('resolvePredictionDialog');
    });
});
