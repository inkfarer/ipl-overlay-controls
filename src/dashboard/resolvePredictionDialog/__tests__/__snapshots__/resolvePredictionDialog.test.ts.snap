// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResolvePredictionDialog matches snapshot when the current round is not completed 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> The current round has not been completed!
  </div>
</div>
<!--v-if-->
<div data-v-d255ffee="" class="ipl-space color-dark m-t-8"><label data-v-820fd73e="" class="ipl-label"> Team A </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-1-button"><span data-v-748defd4="" class="label">First Team</span></button><label data-v-820fd73e="" class="ipl-label winner-label">Winner - Team B </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(255, 199, 0); color: rgb(51, 51, 51);" autocomplete="off" data-test="resolve-outcome-2-button"><span data-v-748defd4="" class="label">Second Team</span></button></div>"
`;

exports[`ResolvePredictionDialog matches snapshot when the prediction is not locked 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-info m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> This prediction cannot be resolved right now.
  </div>
</div>"
`;

exports[`ResolvePredictionDialog matches snapshot when the winner cannot be determined automatically 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> The current round has not been completed!
  </div>
</div>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-info m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> Unable to determine the current round's winner. The leading team is unknown team b.
  </div>
</div>
<div data-v-d255ffee="" class="ipl-space color-dark m-t-8"><label data-v-820fd73e="" class="ipl-label"> Team A </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-1-button"><span data-v-748defd4="" class="label">First Team</span></button><label data-v-820fd73e="" class="ipl-label"> Team B </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-2-button"><span data-v-748defd4="" class="label">Second Team</span></button></div>"
`;

exports[`ResolvePredictionDialog matches snapshot when the winner is determined as team A 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> The current round has not been completed!
  </div>
</div>
<!--v-if-->
<div data-v-d255ffee="" class="ipl-space color-dark m-t-8"><label data-v-820fd73e="" class="ipl-label winner-label">Winner - Team A </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(255, 199, 0); color: rgb(51, 51, 51);" autocomplete="off" data-test="resolve-outcome-1-button"><span data-v-748defd4="" class="label">First Team</span></button><label data-v-820fd73e="" class="ipl-label"> Team B </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-2-button"><span data-v-748defd4="" class="label">Second Team</span></button></div>"
`;

exports[`ResolvePredictionDialog matches snapshot when the winner is determined as team B 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> The current round has not been completed!
  </div>
</div>
<!--v-if-->
<div data-v-d255ffee="" class="ipl-space color-dark m-t-8"><label data-v-820fd73e="" class="ipl-label"> Team A </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-1-button"><span data-v-748defd4="" class="label">First Team</span></button><label data-v-820fd73e="" class="ipl-label winner-label">Winner - Team B </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(255, 199, 0); color: rgb(51, 51, 51);" autocomplete="off" data-test="resolve-outcome-2-button"><span data-v-748defd4="" class="label">Second Team</span></button></div>"
`;

exports[`ResolvePredictionDialog matches snapshot when the winner name cannot be determined 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> The current round has not been completed!
  </div>
</div>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-info m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> Unable to determine the current round's winner.
    <!--v-if-->
  </div>
</div>
<div data-v-d255ffee="" class="ipl-space color-dark m-t-8"><label data-v-820fd73e="" class="ipl-label"> Team A </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-1-button"><span data-v-748defd4="" class="label">First Team</span></button><label data-v-820fd73e="" class="ipl-label"> Team B </label><button data-v-748defd4="" class="ipl-button" style="background-color: rgb(97, 85, 189); color: white;" autocomplete="off" data-test="resolve-outcome-2-button"><span data-v-748defd4="" class="label">Second Team</span></button></div>"
`;

exports[`ResolvePredictionDialog matches snapshot with missing prediction data 1`] = `
"<ipl-dialog-title-stub title="Resolve Prediction" closingdisabled="false"></ipl-dialog-title-stub>
<ipl-error-display-stub class="m-t-8"></ipl-error-display-stub>
<div data-v-c35f7b2a="" class="ipl-message__wrapper ipl-message__type-warning m-t-8">
  <font-awesome-icon-stub data-v-c35f7b2a="" icon="[object Object]" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" class="icon"></font-awesome-icon-stub>
  <div data-v-c35f7b2a="" class="message-content">
    <!----> No prediction data loaded.
  </div>
</div>"
`;
